framework/src/media/voice/SoftwareEndPointDetector.cpp

```diff
@@ -50,5 +61,41 @@ bool SoftwareEndPointDetector::startEndPointDetect(int timeout)
 	return true;
 }
 
+bool SoftwareEndPointDetector::waitEndPoint(int timeout)
+{
+	int ret = -1;
+
+	if (timeout < 0) {
+		ret = sem_wait(&mSem);
+	} else {
+		struct timespec waketime;
+		clock_gettime(CLOCK_REALTIME, &waketime);
+		waketime.tv_sec += timeout;
+		ret = sem_timedwait(&mSem, &waketime);
+	}
+
+	return ret == 0 ? true : false;
+}
+
+bool SoftwareEndPointDetector::detectEndPoint(short *sample, int numSample)
+{
+	int position = 0;
+	while (position + (CONFIG_VOICE_SOFTWARE_EPD_FRAMESIZE << 1) < numSample) {
```

Comment:
Bit operation may added because of it assuming that stereo, am I right?
If I am right, then please add @TODO this(rechanneling) should be handled with audio_manager in future..

Reply:
Actually, bit operation is used because speex handle 16 bit data.
I modified the code to remove bit operation